### PR TITLE
Movielens20m dataset

### PR DESCRIPTION
**Pull request by zkid18**
### Pull Request FAQ
- [documentation](https://catalyst-team.github.io/catalyst/)
- [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)
- [minimal examples section](https://github.com/catalyst-team/catalyst#minimal-examples)
- [changelog](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md) for main framework updates
- [Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion

## Description

Movielens 20M dataset for RecSys.
Current implementation progress:

[x] download dataset
[x] parser and pre-processing for the rating.csv
[x] user-based train/test splitter
[ ] interaction matrix generation
[ ] seq dataset generation 
[ ] upload dataset in sparse format in PyTorch.

I stuck with a proper final representation of the dataset. 

Couple words on Movielens20M dataset. 
1-5 can be considered as context-based information, while `rating` provides the user interaction data. Hence we can omit 1-5 at that time. 

```
    The data are contained in six files: 
    1. genome-scores.csv 
    2. genome-tags.csv 
    3. links.csv 
    4. movies.csv
    5. tags.csv  
    6. ratings.csv
 ```
 
`rating.csv head -5`

```
   userId  movieId  rating   timestamp
0       1        2     3.5  1112486027
1       1       29     3.5  1112484676
2       1       32     3.5  1112484819
3       1       47     3.5  1112484727
4       1       50     3.5  1112484580
```

So I have split the dataset randomly by users. I need to better understand how the dataset for the sequential RecSys should look like. 

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### Checklist
- [ ] Have you updated tests for the new functionality?
- [ ] Have you added your new classes/functions to the docs?
- [ ] Have you updated the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?
- [ ] Have you run [colab minimal CI/CD](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/colab_ci_cd.ipynb) with `latest` and `minimal` requirements?
- [ ] Have you checked XLA integration with [single](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/Catalyst_XLA_single_process.ipynb) and [multiple](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/Catalyst_XLA_multi_process.ipynb) processes?

### CODE REVIEW

File: tmp_data/MovieLens20M/raw/ml-20m/README.txt [hunk 1]
```diff
@@ -0,0 +1,183 @@
+Summary
```
Comment (Scitator) [inline tmp_data/MovieLens20M/raw/ml-20m/README.txt]: not sure about this file ;)

**Review by Scitator — COMMENTED** [review]
it would be great if we could correct the code style and add a few usage examples for the dataset

**Comment by zkid18** [comment]
https://colab.research.google.com/drive/1l6mpAWBKbkEWILnz9ThwpAoTQDCXkG-M?authuser=1#scrollTo=HkzzfT1moccj&uniqifier=1

Colab to test MovieLens20m

File: tmp_data/MovieLens20M/raw/ml-20m/README.txt (hunk 1 above)
Comment (zkid18) [reply tmp_data/MovieLens20M/raw/ml-20m/README.txt]: oops

**Review by Scitator — DISMISSED** [review]
LGTM

**Review by Scitator — DISMISSED** [review]
todo: `torch>=1.3.0` ;)

File: requirements/requirements-cv.txt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 imageio>=2.5.0
 opencv-python-headless>=4.1.1.26
 scikit-image<0.19.0>=0.16.1
-torchvision>=0.4.1
+torchvision=0.5.0
```
Comment (Scitator) [inline requirements/requirements-cv.txt]: 0.4.1

File: requirements/requirements-ml.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
 scipy>=1.4.1
 matplotlib>=3.1.0
-pandas>=0.25.0
+pandas>=1.1.1
```
Comment (Scitator) [inline requirements/requirements-ml.txt]: 1.0?

**Review by Scitator — DISMISSED** [review]

File: requirements/requirements-ml.txt (hunk 3 above)
Comment (Scitator) [reply requirements/requirements-ml.txt]: 1.0? or `0.25.0`

File: tests/catalyst/contrib/datasets/test_movielens_20m.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+        print("Error! Code: {c}, Message, {m}".format(c=type(e).__name__, m=str(e)))
+
+
+@minversion
+@pytest.mark.skipif(not (SETTINGS.ml_required), reason="No catalyst[ml] required")
+def test_download_split_by_user():
+    """
+    Test movielense download
+    """
+    MovieLens20M("./tmp_data", download=True, sample=True)
```
Comment (Scitator) [inline tests/catalyst/contrib/datasets/test_movielens_20m.py:33]: could we use TemporaryDirectory? like https://github.com/catalyst-team/catalyst/blob/e1d78b7dd568c4d5cd94be417d6d8b93ef27b2e5/tests/catalyst/runners/test_train_flags.py

**Review by Scitator — DISMISSED** [review]
comments ;)

**Pull request merged**